### PR TITLE
Add few node helpers

### PR DESCRIPTION
**Pull request by dennwc**
Add a `ChildrenCount` helper (requested by https://github.com/bblfsh/client-go/issues/109) and relax argument constraints for `PositionsOf`, `RolesOf` and `TokenOf` (see https://github.com/bblfsh/client-go/issues/100).

Fixes https://github.com/bblfsh/client-go/issues/109.

### CODE REVIEW

File: uast/nodes/node.go [hunk 1]
```diff
@@ -948,3 +948,28 @@ func UniqueKey(n Node) Comparable {
 		return unkPtr(ptr)
 	}
 }
+
+// ChildrenCount returns a number of immediate children nodes. For arrays it will be the
+// length of an array. For objects, all child Object and Array nodes in fields will be counted.
```
Comment (creachadair) [inline uast/nodes/node.go]: I was confused (until I read the code) about whether this means the Array is counted as 1 (for itself) or for its children. May I suggest

```
// ChildrenCount reports the number of immediate children of n. If n is an Array this is
// the length of the array. If n is an object, each object or array in a field of n counts as 
// one child.
```

or words to that effect?

File: uast/uast.go [hunk 2]
```diff
@@ -182,7 +182,11 @@ func AsPosition(m nodes.Object) *Position {
 }
 
 // PositionsOf returns a complete positions map for the given UAST node.
-func PositionsOf(m nodes.Object) Positions {
+func PositionsOf(n nodes.Node) Positions {
```
Comment (creachadair) [inline uast/uast.go]: Please update the comment to document that this is empty for a non-Object. Previously it was clear from the type signature, but no longer—and intuitively it's not obvious that an array wouldn't also have positions (e.g., the start and end of its span). Not saying it should, but if I asked for the position of an arbitrary node I would be surprised that it should't work for an Array.

File: uast/uast.go [hunk 3]
```diff
@@ -216,7 +220,11 @@ func RoleList(roles ...role.Role) nodes.Array {
 }
 
 // RolesOf is a helper for getting node UAST roles (see KeyRoles).
-func RolesOf(m nodes.Object) role.Roles {
+func RolesOf(n nodes.Node) role.Roles {
+	m, ok := n.(nodes.Object)
```
Comment (creachadair) [inline uast/uast.go]: As 185. Is there some reason for this relaxation more generally?

**Comment by bzz** [comment]
CI seems to be failing for some reason

File: uast/nodes/node.go (hunk 1 above)
Comment (dennwc) [reply uast/nodes/node.go]: Changed the comment, PTAL.

File: uast/uast.go (hunk 2 above)
Comment (dennwc) [reply uast/uast.go]: Done.

File: uast/uast.go (hunk 3 above)
Comment (dennwc) [reply uast/uast.go]: The only reason to relax this requirement is that all XPath query and any manual node manipulations will usually return a `nodes.Node`, and the user have to type assert it each time to use those functions.

**Comment by dennwc** [comment]
CI passes, fixed comments and added test cases. Ready for another round.

**Comment by dennwc** [comment]
@bzz PTAL

**Pull request merged**